### PR TITLE
:arrow_up: Sylius 1.12

### PR DESCRIPTION
**Pull request by elyanory**
Hello team!

A small pull request to allow the update to sylius 1.12 :)

### CODE REVIEW

**Comment by maximehuran** [comment]
Hi @elyanory 

Thank you for your contribution.
Can you check #182 to manage the Makefile and maybe some others files to change ?

File: composer.json [hunk 1]
```diff
@@ -5,10 +5,10 @@
     "description": "A Rich Editor plugin for Sylius.",
     "license": "MIT",
     "require": {
-        "php": "~7.4|~8.0",
+        "php": ">=8.1",
```
Comment (jacquesbh) [inline composer.json:8]: This is a BC, right?
Keeping old versions is mandatory for Sylius 1.8+.

Can you adapt the change please ?

Thank you!!

J

**Review by jacquesbh — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (elyanory) [reply composer.json]: Hi @jacquesbh
I just made the necessary changes 😉

**Comment by elyanory** [comment]
Hi @maximehuran !
I had to make some changes in the .github folder to add php8.1 and I added version 1.12 of sylius in the Makefile.

File: .github/workflows/security.yaml [hunk 2]
```diff
@@ -15,16 +15,16 @@ jobs:
...
 
       - name: Setup PHP
-        run: |
-          sudo update-alternatives --set php /usr/bin/php${{ matrix.php }}
-          echo "date.timezone=UTC" | sudo tee /etc/php/${{ matrix.php }}/cli/conf.d/timezone.ini
-          echo "${{ matrix.php }}" > .php-version
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          extensions: gd, intl, json
```
Comment (maximehuran) [inline .github/workflows/security.yaml:27]:
```suggestion
        uses: shivammathur/setup-php@v2
        with:
          php-version: ${{ matrix.php }}
          extensions: gd, intl, json

      - name: Set project php-version
        run: |
          echo "${{ matrix.php }}" > .php-version
```

File: .github/workflows/recipe.yaml [hunk 3]
```diff
@@ -19,23 +19,24 @@ jobs:
...
       - name: Setup PHP
-        run: |
-          sudo update-alternatives --set php /usr/bin/php${{ matrix.php }}
-          echo "date.timezone=UTC" >> /tmp/timezone.ini
-          sudo mv /tmp/timezone.ini /etc/php/${{ matrix.php }}/cli/conf.d/timezone.ini
-          echo ${{ matrix.php }} > .php-version
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          extensions: gd, intl, json
```
Comment (maximehuran) [inline .github/workflows/recipe.yaml:39]:
```suggestion
        uses: shivammathur/setup-php@v2
        with:
          php-version: ${{ matrix.php }}
          extensions: gd, intl, json

      - name: Set project php-version
        run: |
          echo "${{ matrix.php }}" > .php-version
```

**Review by maximehuran — CHANGES_REQUESTED** [review]

**Review by maximehuran — APPROVED** [review]

**Comment by maximehuran** [comment]
I have an display issue on the admin : 

image

Also if I save an UI Element I have this error : 

image

File: src/UiElement/Registry.php [hunk 4]
```diff
@@ -68,7 +68,7 @@ public function getUiElements(): array
     /**
      * @inheritdoc
      */
-    public function jsonSerialize()
+    public function jsonSerialize(): mixed
```
Comment (jacquesbh) [inline src/UiElement/Registry.php:71]: In that case the PHP version in composer.json is wrong.

😭

**Review by jacquesbh — CHANGES_REQUESTED** [review]

File: src/UiElement/Registry.php (hunk 4 above)
Comment (maximehuran) [reply src/UiElement/Registry.php:71]: What can we can do ?

**Comment by maximehuran** [comment]
What is this commit ? @elyanory 

image

**Review by maximehuran — APPROVED** [review]
SGTM ! I tested the app and it works 🚀

**Review by delyriand — APPROVED** [review]

**Review by jacquesbh — APPROVED** [review]

**Pull request merged**